mobile_verifier/src/seniority.rs

```diff
@@ -0,0 +1,206 @@
+use chrono::{DateTime, Duration, Utc};
+use file_store::file_sink::FileSinkClient;
+use sqlx::{Postgres, Transaction};
+use uuid::Uuid;
+
+use helium_proto::services::poc_mobile as proto;
+
+use crate::heartbeats::{KeyType, ValidatedHeartbeat};
+
+#[derive(Clone, Debug, PartialEq, sqlx::FromRow)]
+pub struct Seniority {
+    pub uuid: Uuid,
+    pub seniority_ts: DateTime<Utc>,
+    pub last_heartbeat: DateTime<Utc>,
+    pub inserted_at: DateTime<Utc>,
+    pub update_reason: i32,
+}
+
+impl Seniority {
+    pub async fn fetch_latest(
+        key: KeyType<'_>,
+        exec: &mut Transaction<'_, Postgres>,
+    ) -> Result<Option<Self>, sqlx::Error> {
+        sqlx::query_as(
+            "SELECT uuid, seniority_ts, last_heartbeat, inserted_at, update_reason FROM seniority WHERE radio_key = $1 ORDER BY last_heartbeat DESC, seniority_ts DESC LIMIT 1",
+        )
+        .bind(key)
+        .fetch_optional(&mut *exec)
+        .await
+    }
+}
+
+pub struct SeniorityUpdate<'a> {
+    key: KeyType<'a>,
+    heartbeat_ts: DateTime<Utc>,
+    uuid: Uuid,
+    pub action: SeniorityUpdateAction,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum SeniorityUpdateAction {
+    NoAction,
+    Insert {
+        new_seniority: DateTime<Utc>,
+        update_reason: proto::SeniorityUpdateReason,
+    },
+    Update {
+        curr_seniority: DateTime<Utc>,
+    },
+}
+
+impl<'a> SeniorityUpdate<'a> {
+    pub fn new(
+        key: KeyType<'a>,
+        heartbeat_ts: DateTime<Utc>,
+        uuid: Uuid,
+        action: SeniorityUpdateAction,
+    ) -> Self {
+        Self {
+            key,
+            heartbeat_ts,
+            uuid,
+            action,
+        }
+    }
+
+    pub fn from(
```

Comment:
you shouldn't have a method named from. That's confusing

Reply:
fixed 15874cb